### PR TITLE
Fix raster geometry updates

### PR DESCRIPTION
**Pull request by TimSylvester**
Handle the case where the raster bucket changes the geometry after a drawable is created from it.

Allow index vector to be empty if there's a buffer Skip drawable create in some cases where there's no data to draw. Reset buffers as well as values when clearing index/vertex containers.  Add more validation. Fixed a bug in layer tweaker update that prevented it from working.

### CODE REVIEW

**Comment by stefankarschti** [comment]
In _legacy_ code the missing indexes on raster `bucket` signals to use the raster quad (`staticDataVertices`, `staticDataIndices` & `staticDataSegments`), so a `drawable` without indexes shouldn't be created in any case.

This condition can vary back & forth during the program execution, the bucket generates buffers only when it decides to clip visible tiles, otherwise leaves them empty so the standard quad is used. [Legacy code here.](https://github.com/maplibre/maplibre-native/blob/9ae0d8f0e41c6fc1800c32020e629a633b976cca/src/mbgl/renderer/layers/render_raster_layer.cpp#L194)

**Comment by TimSylvester** [comment]
The problem is that a drawable is created with masked vertices in the shared data, but the bucket later changes that, or clears the shared vertices.  The segments aren't shared, though, so that won't work.  Or the drawable is set up with the static vertices and the bucket later updates to masked shared vertices which has no effect on the drawable at all.

I considered emplacing the static values into the shared buffers and updating the segments instead, but changing the geometry in a drawable will lead to problems later on.

As an improvement, we should find a way to transfer the textures over, but that's not easy any more.

**Comment by TimSylvester** [comment]
Turns out the buffers can't be reset when `VertexVector`/`IndexVector` is cleared, or it breaks OpenGL.  I believe this indicates a potential bug, in that `SymbolBucket` is changing the values dynamically (`updateBucketOpacities`), but the resulting dirty vectors are not being re-applied in the GL implementation.  I'll create an issue so it doesn't get lost.

File: src/mbgl/renderer/layers/symbol_layer_tweaker.cpp [hunk 1]
```diff
@@ -131,7 +131,7 @@ void SymbolLayerTweaker::execute(LayerGroupBase& layerGroup,
         const auto& symbolData = static_cast<gfx::SymbolDrawableData&>(*drawable.getData());
         const auto isText = (symbolData.symbolType == SymbolType::Text);
 
-        if (isText && (textPaintBuffer || textPropertiesUpdated)) {
+        if (isText && (!textPaintBuffer || textPropertiesUpdated)) {
```
Comment (TimSylvester) [inline src/mbgl/renderer/layers/symbol_layer_tweaker.cpp:134]: Note: This was causing a UBO to be updated every time, and should improve performance.

File: include/mbgl/renderer/layer_tweaker.hpp [hunk 2]
```diff
@@ -93,9 +93,12 @@ class LayerTweaker {
     std::vector<std::string> propertiesAsUniforms;
 #endif // MLN_RENDER_BACKEND_METAL
 
+    // Indicates that the evaluated properties have changed
     bool propertiesUpdated = true;
 
-    bool propertiesChanged = true;
```
Comment (stefankarschti) [inline include/mbgl/renderer/layer_tweaker.hpp:98]: `propertiesChanged` is still referenced. I get a compile error in `LineLayerTweaker` / metal.

Comment (TimSylvester) [reply include/mbgl/renderer/layer_tweaker.hpp:98]: Forgot to try Metal, sorry.

**Review by stefankarschti — APPROVED** [review]
works nicely!

**Review by mwilsnd — APPROVED** [review]

**Pull request merged**